galsim/utilities.py

```diff
@@ -296,6 +296,25 @@ def _convertPositions(pos, units, func):
 
     return pos
 
+def _lin_approx_err(x, f, i):
+    """Error as \int abs(f(x) - approx(x)) when using ith data point to make piecewise linear
+    approximation."""
+    xleft, xright = x[:i+1], x[i:]
+    fleft, fright = f[:i+1], f[i:]
+    xi, fi = x[i], f[i]
+    mleft = (fi-f[0])/(xi-x[0])
+    mright = (f[-1]-fi)/(x[-1]-xi)
+    f2left = f[0]+mleft*(xleft-x[0])
+    f2right = fi+mright*(xright-xi)
+    return np.trapz(np.abs(fleft-f2left), xleft), np.trapz(np.abs(fright-f2right), xright)
+
+def _lin_approx_split(x, f):
+    """Optimally split a tabulated function into two-part piecewise linear approximation.
+    """
+    errs = [_lin_approx_err(x, f, i) for i in xrange(1, len(x)-1)]
```

Comment:
I think it's a bit better than O(N^3) (on average), since the `_lin_approx_split` ranges decrease in size as you go through the outer loop, but yes, it's slow!  I think the absolute values make this hard to simplify (maybe we should think about mean squared error?), but if you think of something definitely do go for it, or let me know and I'll have a go.